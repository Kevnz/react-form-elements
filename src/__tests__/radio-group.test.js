import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'

import RadioGroup from '../radio-group'

describe('Radio component', () => {
  afterEach(() => {
    cleanup()
  })
  describe('The Rendered HTML', () => {
    it('should have a label', () => {
      const { getByText, container } = render(
        <RadioGroup
          name="mygroup"
          label="The Radio"
          initialChecked="first"
          options={[
            { value: 'first', label: 'First' },
            { value: 'second', label: 'Second' },
          ]}
        />
      )

      expect(container.firstChild).toMatchSnapshot()
      const label = getByText('First')
      expect(label.textContent.trim()).toBe('First')
    })
  })
  describe('handling input', () => {
    it('should set a value when passed in', () => {
      const { getByLabelText, container } = render(
        <RadioGroup
          name="mygroup"
          label="The Radio"
          initialChecked="first"
          options={[
            { value: 'first', label: 'First' },
            { value: 'second', label: 'Second' },
          ]}
        />
      )

      expect(container.firstChild).toMatchSnapshot()
      const input = getByLabelText('First')
      expect(input.checked).toBe(true)
    })
    it('should update a value when changed in', () => {
      const { getByLabelText, container } = render(
        <RadioGroup
          name="mygroup"
          label="The Radio"
          initialChecked="first"
          options={[
            { value: 'first', label: 'First' },
            { value: 'second', label: 'Second' },
          ]}
        />
      )

      expect(container.firstChild).toMatchSnapshot()
      const hidden = container.querySelector('input[type="hidden"]')
      expect(hidden.value).toBe('first') // implementation detail, bad expect
      const input = getByLabelText('Second')
      expect(input.checked).toBe(false)
      fireEvent.click(input)
      expect(input.value).toBe('second')
      expect(input.checked).toBe(true)

      expect(hidden.value).toBe('second') // implementation detail, bad expect
    })
  })
})
