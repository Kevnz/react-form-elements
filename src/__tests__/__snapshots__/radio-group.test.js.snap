// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio component The Rendered HTML should have a label 1`] = `
<div
  class="rfe-form__row rfe-radiogroup"
>
  <label>
    The Radio
  </label>
  <div
    class="rfe-radiogroup__row"
  >
    <label
      for="id-1-1"
    >
       
      First
       
    </label>
    <input
      checked=""
      id="id-1-1"
      name="mygroup"
      type="radio"
      value="first"
    />
  </div>
  <div
    class="rfe-radiogroup__row"
  >
    <label
      for="id-2-1"
    >
       
      Second
       
    </label>
    <input
      id="id-2-1"
      name="mygroup"
      type="radio"
      value="second"
    />
  </div>
  <input
    type="hidden"
    value="first"
  />
</div>
`;

exports[`Radio component handling input should set a value when passed in 1`] = `
<div
  class="rfe-form__row rfe-radiogroup"
>
  <label>
    The Radio
  </label>
  <div
    class="rfe-radiogroup__row"
  >
    <label
      for="id-3-1"
    >
       
      First
       
    </label>
    <input
      checked=""
      id="id-3-1"
      name="mygroup"
      type="radio"
      value="first"
    />
  </div>
  <div
    class="rfe-radiogroup__row"
  >
    <label
      for="id-4-1"
    >
       
      Second
       
    </label>
    <input
      id="id-4-1"
      name="mygroup"
      type="radio"
      value="second"
    />
  </div>
  <input
    type="hidden"
    value="first"
  />
</div>
`;

exports[`Radio component handling input should update a value when changed in 1`] = `
<div
  class="rfe-form__row rfe-radiogroup"
>
  <label>
    The Radio
  </label>
  <div
    class="rfe-radiogroup__row"
  >
    <label
      for="id-5-1"
    >
       
      First
       
    </label>
    <input
      checked=""
      id="id-5-1"
      name="mygroup"
      type="radio"
      value="first"
    />
  </div>
  <div
    class="rfe-radiogroup__row"
  >
    <label
      for="id-6-1"
    >
       
      Second
       
    </label>
    <input
      id="id-6-1"
      name="mygroup"
      type="radio"
      value="second"
    />
  </div>
  <input
    type="hidden"
    value="first"
  />
</div>
`;
